addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/KM200Comm.java

```diff
@@ -0,0 +1,1145 @@
+                    sPService.setSetpointProperty(propObject.get("id").getAsString());
+                    newObject.setValueParameter(sPService);
+                    newObject.setJSONData(decodedData);
+                    device.virtualList.add(newObject);
+                    break;
+
+                case "errorList": /* Check whether the type is a errorList */
+                    logger.debug("initDevice: type errorList: {}", decodedData);
+                    KM200ErrorService eService = new KM200ErrorService();
+                    eService.updateErrors(nodeRoot);
+                    newObject.setValueParameter(eService);
+                    newObject.setJSONData(decodedData);
+                    device.virtualList.add(newObject);
+                    break;
+
+                case "refEnum": /* Check whether the type is a refEnum */
+                    logger.debug("initDevice: type refEnum: {}", decodedData);
+                    JsonArray refers = nodeRoot.get("references").getAsJsonArray();
+                    for (int i = 0; i < refers.size(); i++) {
+                        JsonObject subJSON = refers.get(i).getAsJsonObject();
+                        id = subJSON.get("id").getAsString();
+                        initObjects(id, newObject);
+                    }
+                    break;
+
+                case "moduleList": /* Check whether the type is a moduleList */
+                    logger.debug("initDevice: type moduleList: {}", decodedData);
+                    JsonArray vals = nodeRoot.get("values").getAsJsonArray();
+                    for (int i = 0; i < vals.size(); i++) {
+                        JsonObject subJSON = vals.get(i).getAsJsonObject();
+                        id = subJSON.get("id").getAsString();
+
+                        initObjects(id, newObject);
+                    }
+                    break;
+
+                case "yRecording": /* Check whether the type is a yRecording */
+                    logger.debug("initDevice: type yRecording: {}", decodedData);
+                    /* have to be completed */
+                    break;
+
+                case "systeminfo": /* Check whether the type is a systeminfo */
+                    logger.debug("initDevice: type systeminfo: {}", decodedData);
+                    JsonArray sInfo = nodeRoot.get("values").getAsJsonArray();
+                    newObject.setValue(sInfo);
+                    /* have to be completed */
+                    break;
+                case "arrayData":
+                    logger.debug("initDevice: type arrayData: {}", decodedData);
+                    newObject.setJSONData(decodedData);
+                    /* have to be completed */
+                    break;
+
+                case "eMonitoringList":
+                    logger.debug("initDevice: type eMonitoringList: {}", decodedData);
+                    newObject.setJSONData(decodedData);
+                    /* have to be completed */
```

Comment:
There is a TODO, what should be done and can it be completed before merging the PR?

Reply:
It is a missing handling for some features. It is integrated in some heating devices only . This can be done later.